### PR TITLE
MTV plan execution workflow

### PR DESCRIPTION
**Pull request by ydayagi**
FLPATH-1412
https://issues.redhat.com/browse/FLPATH-1412
Create a migration resource for executing a migration plan

### CODE REVIEW

File: mtv-migration/mtv.sw.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          before: PT10S
+    transition: CheckMigrationStatus
+  - name: CheckMigrationStatus
+    type: switch
+    dataConditions:
+      - name: MigrationFailure
+        condition: .getStatusResult.status.conditions | map(select(.category == "Error" or .category == "Critical" or .type == "Failed")) | length > 0
+        transition: MigrationFailure
+      - name: MigrationSuccessful
+        condition: .getStatusResult.status.conditions | map(select(.type == "Succeeded")) | length > 0
```
Comment (gabriel-farache) [inline mtv-migration/mtv.sw.yaml:59]: there is `|length > 0` for both conditions, so, the second one will never match if `length > 0` as the 1st condition will match first no?

Comment (ydayagi) [reply mtv-migration/mtv.sw.yaml:59]: correct. Success conditions will not exist when there is a failure. also the other way around

**Review by jordigilh — APPROVED** [review]
/lgtm

**Review by masayag — APPROVED** [review]

**Pull request merged**